pallets/gas/src/lib.rs

```diff
@@ -327,6 +340,83 @@ pub mod pallet {
 
             Ok(None)
         }
+
+        pub(super) fn create_from_with_value(
+            key: H256,
+            new_node_key: H256,
+            amount: u64,
+            reserve: bool,
```

Comment:
copyed from the former `split_with_value`, but with this `reserve` param to detect which kind of node will be created